app/helpers/quadicon_helper.rb

```diff
@@ -94,7 +94,7 @@ def self.settings_key(klass, layout)
         layout.to_sym
       end
     else
-      klass.base_model.name.underscore.to_sym
+      klass.name.demodulize.underscore.to_sym
```

Comment:
Can't be rather the `base_model` definition changed in the `PhysicalSwitch`? Not sure if possible, just asking.

Reply:
If we change the base model we have problems with the database. We are using the switches table to store the data. If we define, Rails will try to find the `physical_switches` table.

Reply:
I'm just afraid that you're changing this for every model that doesn't fit into the if above, could you please rather do an elsif and keep the else branch if possible?

Reply:
Ok. Give me some time to try and I will update that.

Reply:
Hi @skateman. What about the following code?

```ruby
elsif klass.base_model.name != klass.name.demodulize
  klass.name.demodulize.underscore.to_sym
else
  klass.base_model.name.underscore.to_sym
end
```

I am not sure if that is really necessary, but I will trust your judgement. If you agree with that proposal I will update the PR with that.

Reply:
What if you would store the quadicon setting for the `Switch` model instead? Then you could use the `base_model` and for regular switches it would just fall back to `single_quad` if really necessary.

Reply:
Base switches do not have several information from physical switches as power state and health state. I think we can't move the quadicons to them.

Reply:
Not the quadicons, just the setting for having a quad or not.

Reply:
Hi @skateman, I was playing around with your proposal, but I would like your opinion about something. In order to move the configuration to Switches, I need to change the configuration at `app/views/configuration/_ui_1.html.haml`. It is a good idea to add the role checking, but there is no role defined for Switches yet, only for Physical Switches. I don't think it is a good idea to create roles for Switches right now because we will have the features and the roles in the Core project, and there is no project using them. Also, we will have to check both permissions every time when we want to access Physical Switches (if the user has permissions for Switches and for Physical Switches)

Another solution is to use `true` in the role checking value as we have for `VM`, but the option will always be there. And we will have the problem stated before if someone update that line to have a role to control switches in another part of the project.

The second solution is easy, but I am not sure if it is a good idea if other providers start to use Switches in some place. Adding a role there will break the Physical Switches quadicons.

Thanks in advance. I will wait for your comments about.

Reply:
I see no problem in using the role associated to physical switches when it's just about displaying the UI element. The switches have no roles and not even quadicons, so you're completely safe here. Anyway this screen is growing with new datatypes and should be redesigned/refactored a little bit, so you don't even have to feel bad for adding extra complexity. Just please add a comment here that the model is the same, but the role is different and you're good to go :wink:

Reply:
As we discussed at Gitter today earlier, I will wait for your return before updating the code. Here are some outputs that can be helpful:

```
irb(main):007:0> Switch.base_model.name
=> "Switch"
irb(main):008:0> PhysicalSwitch.base_model.name
=> "Switch"
irb(main):009:0> Switch.name
=> "Switch"
irb(main):010:0> Switch.name.demodulize
=> "Switch"
irb(main):011:0> PhysicalSwitch.name.demodulize
=> "PhysicalSwitch"
irb(main):012:0> PhysicalServer.name.demodulize
=> "PhysicalServer"
```

Reply:
@saulotoledo it's not always the same, demodulize basically removes the `Module::` prefixes and it'ts totally allowed to have a `base_model` that has a `Module::` prefix is totally possible.

Reply:
@skateman I've added the ugly `elsif` for now then, and we can handle that in the future as we agreed. Let me know if there are some other changes to do.